### PR TITLE
move to scala 3

### PR DESCRIPTION
**Pull request by bardurdam**
- cross build for scala 2.13 and scala 3
- implement generic derivation without shapeless
- misc adjustments wrt scala 3
- add de locale in sample backend
- add roundtrip tests for derived decoder/encoder
- remove simulacrum usage and add summoner on companion

### CODE REVIEW

**Comment by bardurdam** [comment]
@taig WDYT?

**Comment by bardurdam** [comment]
@taig I ran scalafmt and force pushed

**Comment by taig** [comment]
Thank you for working on this, looks great at first glance! I'll review it as soon as I can

**Comment by ahjohannessen** [comment]
@bardurdam DS_Store is included

File: modules/sample/backend/src/main/scala/io/taig/babel/sample/backend/BackendApp.scala [hunk 1]
```diff
@@ -40,6 +40,7 @@ final class SampleRoutes[F[_]: Sync](i18ns: NonEmptyTranslations[I18n]) extends
     case GET -> Root =>
       val i18n = i18ns(locale)
 
+      // Sync[F].pure(println(i18n,)) >>
```
Comment (ahjohannessen) [inline modules/sample/backend/src/main/scala/io/taig/babel/sample/backend/BackendApp.scala:43]: Leftover

**Comment by bardurdam** [comment]
Removed .DS_Store and added correctly to gitignore. Also removed leftover code.

**Comment by taig** [comment]
Excellent work, thank you. I'll publish a new release shortly

**Comment by ahjohannessen** [comment]
👍

**Pull request merged**